src/Discord.Net.Rest/DiscordRestApiClient.cs

```diff
@@ -75,15 +75,16 @@ internal void SetBaseUrl(string baseUrl)
             RestClient = _restClientProvider(baseUrl);
             RestClient.SetHeader("accept", "*/*");
             RestClient.SetHeader("user-agent", UserAgent);
-            RestClient.SetHeader("authorization", GetPrefixedToken(AuthTokenType, AuthToken));
+            if (!string.IsNullOrEmpty(AuthToken))
+                RestClient.SetHeader("authorization", GetPrefixedToken(AuthTokenType, AuthToken));
         }
         /// <exception cref="ArgumentException">Unknown OAuth token type.</exception>
         internal static string GetPrefixedToken(TokenType tokenType, string token)
         {
             return tokenType switch
             {
                 TokenType.Bot => $"Bot {token}",
-                TokenType.Bearer => $"Bearer {token}",
+                TokenType.Bearer => $"{token}",  // Bearer tokens are not prefixed
```

Comment:
Is this correct? as per [the discord api docs](https://discord.com/developers/docs/reference#authentication), the authorization header follows this format: `Authorization: TOKEN_TYPE TOKEN`.

Also, changing this may break other usecases of bearer tokens

Reply:
I don't know whether it is technically correct. What I do know is that:

- The browser on discord.com/app uses an unprefixed token (with the v9 api)
- Doing this made HTTP errors (401 Unauthorized provided my notes are correct) go away
- I have not personally noticed any ill effects from this change nearly 4 weeks later

Am I using an incorrect token? I was following [this StackOverflow post](https://stackoverflow.com/questions/72605540/getting-bearer-token-for-discord-api).

Reply:
That kind of token is a User Account token, which per the discord TOS is not allowed to be used by 3rd party tools (such as your code and this library).

Reply:
I'll be honest, I hadn't messed with the API directly much. I'm trying to use discord as an authentication provider and then call the API with the subsequent claims/tokens

Reply:
> That kind of token is a User Account token, which per the discord TOS is not allowed to be used by 3rd party tools (such as your code and this library).

Thanks. I'll take a second look when I'm able to circle back to this again.

Reply:
@quinchs where are you seeing this in the ToS? As far as I can tell, it's valid to use a user token to hit the API so long as it's the user initiating the action/it's not automated. 

> Automating normal user accounts (generally called "self-bots") outside of the OAuth2/bot API is forbidden, and can result in account termination if found.

Perhaps it'd be best to add a new token type/a different rest client if the library was to support that.

Reply:
Lookup the term "self botting", that's what you're attempting to do, and it's not allowed, hence the reason they constantly update the API to disallow user tokens from most endpoints

Reply:
@SylveonDeko I'm not automating anything - the user authenticates via discord, and I need to retrieve a list of their guilds so they can choose one for some of my bots functionality. This is happening inside of a web app, not inside of a bot command. If there's some other way I'm supposed to accomplish this I'm all ears, but I fail to see how that's "self botting"

Reply:
so in short if one wants this fixed one has to fork all of this and do it themself?